### PR TITLE
Test Julia 1.7

### PR DESCRIPTION
**Pull request by MichaelHatherly**
Since the package compat bounds are 1.7 that version should be tested in CI.

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -21,6 +21,7 @@ jobs:
     strategy:
       matrix:
         version:
+          - '1.7'
```
Comment (devmotion) [inline .github/workflows/CI.yml:24]: A bit more future-proof:
```suggestion
          - 'min'
```

Comment (MichaelHatherly) [reply .github/workflows/CI.yml:24]: Ah, nice, quite new that feature.

**Comment by MichaelHatherly** [comment]
Skipping changelog check since this isn't really user-visible.

File: Project.toml [hunk 2]
```diff
@@ -27,7 +27,7 @@ MultipleTesting = "0.5, 0.6"
 NaturalSort = "1.0"
 OrderedCollections = "1"
 Printf = "1"
-SHA = "0.7"
+SHA = "0.7, 1.7"
```
Comment (devmotion) [inline Project.toml:30]: Based on https://discourse.julialang.org/t/psa-compat-requirements-in-the-general-registry-are-changing/104958#update-november-9th-2023-2 (and in particular the update from 9 November) I think the safest approach would be to allow
```suggestion
SHA = "<0.0.1, 0.7, 1"
```

**Review by devmotion — APPROVED** [review]

**Pull request merged**